yarn-project/sequencer-client/src/publisher/l1-publisher.ts

```diff
@@ -111,9 +104,17 @@ export type MetadataForSlot = {
  * Adapted from https://github.com/AztecProtocol/aztec2-internal/blob/master/falafel/src/rollup_publisher.ts.
  */
 export class L1Publisher {
+  // @note  If we want to simulate in the future, we have to skip the viem simulations and use `reads` instead
+  //        This is because the viem simulations are not able to simulate the future, only the current state.
+  //        This means that we will be simulating as if `block.timestamp` is the same for the next block
+  //        as for the last block.
+  //        Nevertheless, it can be quite useful for figuring out why exactly the transaction is failing
+  //        as a middle ground right now, we will be skipping the simulation and just sending the transaction
+  //        but only after we have done a successful run of the `validateHeader` for the timestamp in the future.
+  public static SKIP_SIMULATION = true;
```

Comment:
Should this be an env var?

Reply:
Thought about it but not sure. It is not something we really want to turn "off" by mistake as that would cause some of the failing issues that for example @Maddiaa0 have encountered. The main reason I kept it here was such that it was not needed to add the `simulate` if one really wanted to simulate those calls.